### PR TITLE
Feature/common interceptors

### PR DESCRIPTION
**Pull request by LukasFritzeDev**
To make it easy to opt in to some API mechanisms, a few commonly used [interceptors](https://axios-http.com/docs/interceptors) are made easily available.

### CODE REVIEW

File: packages/commons/src/interceptors/accessToken.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    if (!conf.headers.has(headerName)) {
+      conf.headers.set(headerName, token);
+    }
+    return conf;
+  });
+}
+
+export function withAccessToken<T extends ApiClientBase | AxiosInstance>(
+  client: T,
+  token: string | undefined,
```
Comment (mfal) [inline packages/commons/src/interceptors/accessToken.ts:19]: Why optional (and why in `configureAccessTokenInterceptor` not)?

File: packages/commons/src/interceptors/consistencyHandling.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    if (lastEventId !== undefined && !isMutatingRequest) {
+      config.headers["if-event-reached"] = lastEventId;
+    }
+    return config;
+  });
+
+  axios.interceptors.response.use((response) => {
+    const isMutatingRequest = ["post", "put", "delete", "patch"].includes(
+      response.config.method?.toLowerCase() ?? "",
+    );
```
Comment (mfal) [inline packages/commons/src/interceptors/consistencyHandling.ts:22]: Duplicate code (greetings Uncle Bob)

Comment (LukasFritzeDev) [reply packages/commons/src/interceptors/consistencyHandling.ts:22]: Oh. I was sure I had extracted this to a function … 🙈

File: packages/commons/src/interceptors/accessToken.ts (hunk 1 above)
Comment (LukasFritzeDev) [reply packages/commons/src/interceptors/accessToken.ts:19]: This makes it easier to use the function, i.e. in https://github.com/mittwald/api-client-js/pull/45/files#diff-e3af516cc4c91c89a315a806b8921d6cc79ab98f800f1954acf04fd41f5f619c

Comment (LukasFritzeDev) [reply packages/commons/src/interceptors/accessToken.ts:19]: `configureAccessTokenInterceptor` should not be exported

File: packages/commons/src/interceptors/accessToken.ts
[the remark this answers is not in the document]
Comment (LukasFritzeDev) [reply packages/commons/src/interceptors/accessToken.ts:1]: https://github.com/mittwald/api-client-js/issues/46

**Pull request merged**